packages/shared/components/popups/ActivityDetailsPopup.svelte

```diff
@@ -0,0 +1,95 @@
+<script lang="typescript">
+    import { Pill, ActivityTypePill, Box, AddressBox, KeyValueBox } from 'shared/components/atoms'
+    import { Text, Button } from 'shared/components'
+    import { convertToFiat, currencies, exchangeRates, formatCurrency } from 'shared/lib/currency'
+    import { formatDate, localize } from '@core/i18n'
+    import { getOfficialExplorer } from 'shared/lib/network'
+    import { Platform } from 'shared/lib/platform'
+    import { activeProfile } from 'shared/lib/profile'
+    import { CurrencyTypes } from 'shared/lib/typings/currency'
+    import { Payload } from 'shared/lib/typings/message'
+    import { formatUnitBestMatch } from 'shared/lib/units'
+    import { getMilestoneMessageValue, wallet } from 'shared/lib/wallet'
+    import { FontWeightText } from 'shared/components/Text.svelte'
+    import { getTransactionSubjectAddressOrAccount } from '@lib/utils/transactionObject'
+
+    export let id: string
+    export let timestamp: string
+    export let confirmed: boolean
+    export let payload: Payload
+    export let balance: number // migration tx
+
+    let date = localize('error.invalidDate')
+    $: {
```

Comment:
I think this can be defined as a function and called reactively upon change of timestamp.

Reply:
Can do, do you think a util function or just a local one?

Reply:
Maybe util one makes more sense

Reply:
This has moved in https://github.com/iotaledger/firefly/pull/3074 please add this comment there